ddl/db_partition_test.go

```diff
@@ -3615,3 +3615,19 @@ func TestDuplicatePartitionNames(t *testing.T) {
 		"(PARTITION `p2` VALUES IN (2),\n" +
 		" PARTITION `p3` VALUES IN (3))"))
 }
+
+func TestPartitionTableWithAnsiQuotes(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create database partitionWithAnsiQuotes")
+	defer tk.MustExec("drop database partitionWithAnsiQuotes")
+	tk.MustExec("use partitionWithAnsiQuotes")
+	tk.MustExec("SET SESSION sql_mode='ANSI_QUOTES'")
+	tk.MustExec(`create table t1(created_at datetime) PARTITION BY RANGE COLUMNS(created_at) (
+		PARTITION p0 VALUES LESS THAN ('2021-12-01 00:00:00'),
+		PARTITION p1 VALUES LESS THAN ('2022-01-01 00:00:00'))`)
+	tk.MustExec(`create table t2(created_at timestamp) PARTITION BY RANGE (unix_timestamp(created_at)) (
+		PARTITION p0 VALUES LESS THAN (unix_timestamp('2021-12-01 00:00:00')),
+		PARTITION p1 VALUES LESS THAN (unix_timestamp('2022-01-01 00:00:00')))`)
+}
```

Comment:
```suggestion
       t1Def := `CREATE TABLE "t1" (` + "\n" +
               `  "created_at" datetime DEFAULT NULL` + "\n" +
               `) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin` + "\n" +
               `PARTITION BY RANGE COLUMNS("created_at")` + "\n" +
               `(PARTITION "p0" VALUES LESS THAN ('2021-12-01 00:00:00'),` + "\n" +
               ` PARTITION "p1" VALUES LESS THAN ('2022-01-01 00:00:00'))`
       tk.MustQuery(`show create table t1`).Check(testkit.Rows("t1 " + t1Def))
       tk.MustExec(`drop table t1`)
       tk.MustExec(t1Def)
       t2Def := `CREATE TABLE "t2" (` + "\n" +
               `  "created_at" timestamp NULL DEFAULT NULL` + "\n" +
               ") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin\n" +
               "PARTITION BY RANGE (UNIX_TIMESTAMP(`created_at`))\n" +
               `(PARTITION "p0" VALUES LESS THAN (1638313200),` + "\n" +
               ` PARTITION "p1" VALUES LESS THAN (1640991600))`
       tk.MustQuery(`show create table t2`).Check(testkit.Rows("t2 " + t2Def))
       tk.MustExec(`drop table t2`)
       tk.MustExec(t2Def)
}
```
Maybe also check the output under `ANSI_QUOTES` as well as test it as input?
Found https://github.com/pingcap/tidb/issues/35389 while testing :)

Reply:
Let's do it in another PR if it deserves to be fixed.

Reply:
I agree that the output fix should be a separate issue and PR, that is also why I created the new issue #35389